### PR TITLE
add buildmodelmanifest and postbuildmodelmanifest in buildversioninfo

### PR DESCRIPTION
**Pull request by ansyral**
@vwxyzh @superyyrrzz  @hellosnow @qinezh @chenkennt

### CODE REVIEW

**Comment by superyyrrzz** [comment]
> namespace Microsoft.DocAsCode.Build.Engine.Incrementals

copyright header

---
Refers to: src/Microsoft.DocAsCode.Build.Engine/Incrementals/ModelManifest.cs:1 in be8c540. [](commit_id = be8c540ea006a77c69e13883e7d2847b0fe88a8b, deletion_comment = False)

File: src/Microsoft.DocAsCode.Build.Engine/HostService.cs [hunk 1]
```diff
@@ -472,22 +472,33 @@ where changes.Contains(k)
...
+                if (pair.Value == LoadPhase.None)
                 {
-                    // processor.SaveIntermediateModel(f, stream);
-                    // copy files according to lastbuildinfo
-                    // update currentbuildinfo
+                    if (lmm == null)
+                    {
+                        throw new InvalidDataException($"Full build hasn't loaded model {pair.Key.FullPath}");
+                    }
+                    File.Copy(Path.Combine(lmm.BaseDir, fileName), Path.Combine(cmm.BaseDir, fileName));
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: >fileName [](start = 56, length = 8)

question: the intermediate file is without extension?

File: src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs [hunk 2]
```diff
@@ -87,10 +87,14 @@ private Manifest BuildCore(DocumentBuildParameters parameters)
                         DependencyFile = Path.Combine(IntermediateFolder, "dependency"),
                         ManifestFile = Path.Combine(IntermediateFolder, "manifest"),
                         XRefSpecMapFile = Path.Combine(IntermediateFolder, "xrefspecmap"),
+                        BuildModelManifestFile = Path.Combine(IntermediateFolder, "buildmodelmanifest"),
+                        PostBuildModelManifestFile = Path.Combine(IntermediateFolder, "postbuildmodelmanifest"),
                         Attributes = fileAttributes,
                         Dependency = context.DependencyGraph,
                         Manifest = context.ManifestItems,
                         XRefSpecMap = context.XRefSpecMap,
+                        BuildModelManifest = new ModelManifest { BaseDir = CreateRandomDir(IntermediateFolder), Models = new List<string>() },
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: >new List<string>() [](start = 121, length = 18)

can initialize in the property definition?

File: src/Microsoft.DocAsCode.Build.Engine/HostService.cs (hunk 1 above)
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: yes, the files are for reload, extension is not necessary

---
In reply to: [77566882](https://github.com/dotnet/docfx/pull/670#discussion_r77566882) [](ancestors = 77566882)

Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: what if index.json and index.md exist the same time?

---
In reply to: [77568436](https://github.com/dotnet/docfx/pull/670#discussion_r77568436) [](ancestors = 77568436,77566882)

Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: nice question~ that would be a problem

---
In reply to: [77568739](https://github.com/dotnet/docfx/pull/670#discussion_r77568739) [](ancestors = 77568739,77568436,77566882)

**Comment by ansyral** [comment]
#449

**Comment by vwxyzh** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/HostService.cs [hunk 3]
```diff
@@ -623,6 +655,7 @@ internal enum LoadPhase
     {
         None,
         PreBuild,
-        PostBuild
+        PostBuild,
+        PostPostBuild,
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: >PostPostBuild [](start = 8, length = 13)

What is PostPostBuild?

Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: requesting for better name:)

---
In reply to: [77579915](https://github.com/dotnet/docfx/pull/670#discussion_r77579915) [](ancestors = 77579915)

Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: What does this phase do?

---
In reply to: [77580084](https://github.com/dotnet/docfx/pull/670#discussion_r77580084) [](ancestors = 77580084,77579915)

Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: i want to mean the model loaded after "PostBuild" step

---
In reply to: [77627790](https://github.com/dotnet/docfx/pull/670#discussion_r77627790) [](ancestors = 77627790,77580084,77579915)

**Pull request merged**